src/cluster_legacy.c

```diff
@@ -3874,7 +3874,12 @@ void clusterSendMessage(clusterLink *link, clusterMsgSendBlock *msgblock) {
     server.stat_cluster_links_memory += sizeof(listNode);
 
     /* Populate sent messages stats. */
-    uint16_t type = ntohs(getMessageFromSendBlock(msgblock)->type);
+    uint16_t type;
+    if (is_light_hdr) {
+        type = ntohs(getLightMessageFromSendBlock(msgblock)->type) & ~CLUSTERMSG_MODIFIER_MASK;
+    } else {
+        type = ntohs(getMessageFromSendBlock(msgblock)->type);
+    }
```

Comment:
> @zuiderkwast let me know if we can get the type without modifying the method signature.

The `clusterMsgLight` and `clusterMsg` have their `type` field at the same offset, so we should be able to get the type without caring if it's light or not. For stats, we can just clear the light bit.

```suggestion
    /* Ignore the difference between light and heavy message variants. */
    uint16_t type = ntohs(getMessageFromSendBlock(msgblock)->type);
    type &= ~CLUSTERMSG_MODIFIER_MASK;
```